src/fair/allocation.py

```diff
@@ -24,39 +31,71 @@ def initialize_allocation_matrix(items, agents):
     return X
 
 
-def initialize_exchange_graph(items):
+def initialize_edge_matrix(N: int):
+    """Initialize array that keeps track of the agents responsible for a specific edge between two
+    different items on the exchange graph.
+    Initially, there are no edges, thus edge matrix is a len(items) x len(items) array of empty lists.
+
+    Args:
+       N (int): Number of items
+
+    Returns:
+        list: N x N list of empty lists
+    """
+    edge_matrix = []
+    for i in range(N):
+        edge_row = []
+        for j in range(N):
+            edge_row.append([])
+        edge_matrix.append(edge_row)
+    return edge_matrix
+
+
+def initialize_exchange_graph(N: int):
+    """Generate exchange graph. There is one node for every item and a sink node 't' representing the pile of unnasigned items.
+    Initially, there are no edges between items, and an edge from every item node to node 't'.
+    Disclaimer: The previous assumes that every items has capacity > 0
+
+    Args:
+        N (int): number of items
+
+    Returns:
+        nx.graph: networkx graph object
+    """
     exchange_graph = nx.DiGraph()
-    for i in range(len(items)):
+    for i in range(N):
         exchange_graph.add_node(i)
     exchange_graph.add_node("t")
-    for i in range(len(items)):
+    for i in range(N):
         exchange_graph.add_edge(i, "t")
     return exchange_graph
 
 
-def get_max_items(items):
-    max_items = 0
-    for i in range(len(items)):
-        max_items += items[i].capacity
-    return max_items
-
-
-"""Pick agent functions"""
-
+"""Retrieve/update information"""
 
-def pick_agent(X, max_items, items, agents, players):
-    max_capacity = max_items
-    for player in players:
-        agent = agents[player]
-        bundle = get_bundle_from_allocation_matrix(X, items, player)
-        current_utility = agent.valuation(bundle)
-        if current_utility < max_capacity:
-            max_capacity = current_utility
-            agent_picked = player
-    return agent_picked
 
-
-def get_gain_function(X, agents, items, agent_picked, criteria, weights):
+def get_gain_function(
+    X: type[np.ndarray],
+    agents: list[BaseAgent],
+    items: list[ScheduleItem],
+    agent_picked: int,
+    criteria: str,
+    weights: list[float],
+):
+    """Get gain function for a certain agent from the current allocation, according to the gain function criteria and agent assigned weight.
+    This function is used to update the gain function for the item that just played, to keep track of priority agents.
+
```

Comment:
"item that just played" or "agent that just played"?

Reply:
yesss good catch. Changed!